Source/Selling/Mixins/Commodity.lua

```diff
@@ -0,0 +1,130 @@
+AuctionatorCommoditySellingMixin = {}
+
+local AUCTIONATOR_COMMODITY_EVENTS = {
+  "COMMODITY_SEARCH_RESULTS_UPDATED",
+  "COMMODITY_SEARCH_RESULTS_ADDED"
+}
+
+function AuctionatorCommoditySellingMixin:Initialize()
+  Auctionator.Debug.Message("AuctionatorCommoditySellingMixin:Initialize()")
+
+  hooksecurefunc(
+    AuctionHouseFrame.CommoditiesSellFrame.ItemDisplay,
+    "SetItemInternal",
+    function()
+      self:SetDuration(
+        AuctionHouseFrame.CommoditiesSellFrame.DurationDropDown,
+        tonumber(Auctionator.Config.Get(Auctionator.Config.Options.COMMODITY_AUCTION_DURATION))
```

Comment:
`tonumber` is redundant

Reply:
I thought so too, but I was definitely running into issues when executing this code because of this... May have been previously stored values while I was messing with config panels? I'll remove and see what happens...

Reply:
Looking at my saved variables:
```lua
        ["item_auction_duration"] = "24",
        ["auction_item_duration"] = "24",
        ["shift_stack_tooltips"] = true,
        ["show_lists"] = true,
        ["commodity_undercut_percentage"] = 1,
        ["enchant_tooltips"] = true,
        ["auction_tooltips"] = true,
        ["commodity_auction_duration"] = "12",
        ["debug"] = true,
        ["vendor_tooltips"] = true,
        ["mailbox_tooltips"] = true,
        ["commodity_undercut_static_value"] = 500,
        ["auto_list_search"] = true,
        ["autoscan"] = true,
        ["auction_commodity_duration"] = "12",
        ["item_undercut_percentage"] = 5,
        ["auction_duration"] = 24,
        ["commodity_auction_sales_preference"] = "percentage",
        ["item_auction_sales_preference"] = "percentage",
        ["item_undercut_static_value"] = 500,
        ["undercut_percentage"] = 5,
```
There's a lot of shit I can prune, but the duration values are stored as strings somehow...

Reply:
I changed the `KeyValue`s being passed to the radio buttons to `number` from `string` 🙄